emu_sv/sv_config.py

```diff
@@ -115,21 +117,37 @@ def _expected_kwargs(self) -> set[str]:
 
     def monkeypatch_observables(self) -> None:
         obs_list = []
+
         for _, obs in enumerate(self.observables):  # monkey patch
             obs_copy = copy.deepcopy(obs)
+
             if isinstance(obs, Occupation):
-                obs_copy.apply = MethodType(  # type: ignore[method-assign]
-                    qubit_occupation_sv_impl, obs_copy
-                )
-            elif isinstance(obs, EnergyVariance):
+                if self.noise_model.noise_types == ():
+                    obs_copy.apply = MethodType(  # type: ignore[method-assign]
+                        qubit_occupation_sv_impl, obs_copy
+                    )
```

Comment:
what do you think about:

```python
if self.noise_model.noise_types:
    raise NotImplementedError( ...)
```

Reply:
I don't think this comment is at quite the right line of code, but Kemal has a valid point. Somewhere in the `SVConfig` there should be a check that only effective noise is present in the `NoiseModel`